user_groups: Include group_id in success response on group creation.

Aditya8840 opened the pull request:
### Summary
Previously, the `group_id` was not included in the success response of the user group creation API. This omission made it challenging for clients to reference the newly created user group.

### Changes Introduced
This commit enhances the API by returning a success response that includes the unique ID of the user group with the key `group_id`. This improvement simplifies the process for clients to easily reference the newly created user group.

### Fixes
Fixes: #29686

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on zerver/openapi/zulip.yaml:
```diff
@@ -20463,7 +20463,30 @@ paths:
+                      group_id:
+                        type: integer
+                        description: |
+                          The unique ID of the created user groups.
```
"created user group" -- plural is wrong here. One should always carefully review text in any documentation you write, since it looks very unprofessional to have typos in docs.

timabbott commented on zerver/openapi/zulip.yaml:
```diff
@@ -20463,7 +20463,30 @@ paths:
+          description: |
+            A success response containing the unique ID of the user group
+            with the key `group_id`. This field provides a straightforward
+            way to reference the newly created user group.
```
Generally you don't need to mention the name of the key here; that's in the generated documentation

timabbott commented:
Can you post a screenshot of what the documentation looks like?

Aditya8840 commented:
<img width="769" alt="Screenshot 2024-10-28 at 1 29 20 PM" src="https://github.com/user-attachments/assets/a4df6063-cab2-46d5-9d49-e41ae2159c71">

I've addressed the feedback provided. Could you please review it?

timabbott commented:
Looks good, DM me when you've renumbered and I can merge it. Thanks @Aditya8840!

timabbott commented:
Merged, thanks @Aditya8840!
